docs/reference/bqm.rst

```diff
@@ -0,0 +1,161 @@
+.. _bqm:
+
+=======================
+Binary Quadratic Models
+=======================
+
+.. todo: probably we'll want a shorter summary
```

Comment:
I think the current docstring of dimod/binary_quadratic_model.py makes sense here, it would be strange not to have the Ising and QUBO forms in dimod

Reply:
I might want to make it a more generic docstring that defines a BQM and then explains how QUBO/Ising are special cases. But for now I agree it's pretty good.